### PR TITLE
feat: ` avm/res/container-service/managed-cluster` add additional outbound types 

### LINKED ISSUE #897: [AVM Module Issue]: Support outboundType `userAssignedNATGateway` and `managedNATGateway`
### Check for previous/existing GitHub issues

- [X] I have checked for previous/existing GitHub issues

### Issue Type?

Feature Request

### Module Name

avm/res/container-service/managed-cluster

### (Optional) Module Name if not listed above

_No response_

### (Optional) Module Version

_No response_

### Description

Support outboundType `userAssignedNATGateway` and `managedNATGateway`

https://learn.microsoft.com/en-us/azure/aks/nat-gateway

Before:
```bicep
@description('Optional. Specifies outbound (egress) routing method. - loadBalancer or userDefinedRouting.')
@allowed([
  'loadBalancer'
  'userDefinedRouting'
])
param outboundType string = 'loadBalancer'
```

After:
```bicep
@description('Optional. Specifies outbound (egress) routing method.') // Modification
@allowed([
  'loadBalancer'
  'userDefinedRouting'
  'managedNATGateway' // Added
  'userAssignedNATGateway' // Added
])
param outboundType string = 'loadBalancer'
```

### (Optional) Correlation Id

_No response_

### PR DESCRIPTION
**Pull request by ilhaan**
## Description

Adding two new outbound types to the existing AKS module. 

[avm.res.container-service.managed-cluster](https://github.com/ilhaan/bicep-registry-modules/actions/workflows/avm.res.container-service.managed-cluster.yml)

### CODE REVIEW

File: avm/res/container-service/managed-cluster/main.bicep [hunk 1]
```diff
@@ -64,6 +64,8 @@ param managedOutboundIPCount int = 0
 @allowed([
   'loadBalancer'
   'userDefinedRouting'
+  'managedNATGateway'
```
Comment (coolhome) [inline avm/res/container-service/managed-cluster/main.bicep:67]: The description for this property no longer reflects the new egress methods

Comment (ilhaan) [reply avm/res/container-service/managed-cluster/main.bicep:67]: Good catch, fixing now.

Comment (ilhaan) [reply avm/res/container-service/managed-cluster/main.bicep:67]: Fixed: https://github.com/Azure/bicep-registry-modules/pull/992/commits/48fc32cc1aa01ba9ffa00add179ed22fed8f073d

**Comment by AlexanderSehr** [comment]
Hey @JPEasier could you review this PR as the co-owner? :)

**Comment by JPEasier** [comment]
@ilhaan could you add a successful run from your branch to the PR description? then i will merge the PR :)

**Review by JPEasier — CHANGES_REQUESTED** [review]
@ilhaan could you add a successful run from your branch to the PR description? then i will merge the PR :)

**Comment by ChrisSidebotham** [comment]
@ilhaan - Please take a look at the above comments :)

**Comment by ilhaan** [comment]
> @ilhaan could you add a successful run from your branch to the PR description? then i will merge the PR :)

@JPEasier Added the status badge and started a new pipeline run.

**Comment by ilhaan** [comment]
@JPEasier the module is failing these test in the main branch as well: https://github.com/ilhaan/bicep-registry-modules/actions/runs/8210379598

**Comment by ilhaan** [comment]
@JPEasier pipeline is now passing

**Review by JPEasier — APPROVED** [review]

**Pull request merged**